### PR TITLE
No more logger factories in Storage package

### PR DESCRIPTION
**Pull request by agr**
Code changes for the [#3889](https://github.com/NuGet/NuGetGallery/issues/3889)

### CODE REVIEW

File: src/NuGet.Services.Storage/AggregateStorageFactory.cs [hunk 1]
```diff
@@ -10,21 +10,21 @@ namespace NuGet.Services.Storage
     public class AggregateStorageFactory : StorageFactory
     {
         private readonly AggregateStorage.WriteSecondaryStorageContentInterceptor _writeSecondaryStorageContentInterceptor;
-        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger<AggregateStorage> _aggregateStorageFactory;
```
Comment (scottbommarito) [inline src/NuGet.Services.Storage/AggregateStorageFactory.cs]: `_aggregateStorageLogger`

Comment (agr) [inline src/NuGet.Services.Storage/AggregateStorageFactory.cs]: Done

File: src/NuGet.Services.Storage/AggregateStorageFactory.cs [hunk 2]
```diff
@@ -10,21 +10,21 @@ namespace NuGet.Services.Storage
...
+        public AggregateStorageFactory(StorageFactory primaryStorageFactory, ICollection<StorageFactory> secondaryStorageFactories, ILogger<AggregateStorage> aggregateStorageCursor)
+            : this(primaryStorageFactory, secondaryStorageFactories, null, aggregateStorageCursor)
         {
         }
 
-        public AggregateStorageFactory(StorageFactory primaryStorageFactory, StorageFactory[] secondaryStorageFactories,
+        public AggregateStorageFactory(StorageFactory primaryStorageFactory, ICollection<StorageFactory> secondaryStorageFactories,
             AggregateStorage.WriteSecondaryStorageContentInterceptor writeSecondaryStorageContentInterceptor,
-            ILoggerFactory loggerFactory)
+            ILogger<AggregateStorage> aggregateStorageFactory)
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/AggregateStorageFactory.cs]: just call this logger

File: src/NuGet.Services.Storage/AzureStorageFactory.cs [hunk 3]
```diff
@@ -13,14 +13,14 @@ public class AzureStorageFactory : StorageFactory
...
+        private readonly ILogger<AzureStorage> _azureStorageLogger;
 
-        public AzureStorageFactory(CloudStorageAccount account, string containerName, ILoggerFactory loggerFactory, string path = null, Uri baseAddress = null)
+        public AzureStorageFactory(CloudStorageAccount account, string containerName, ILogger<AzureStorage> azureStorageLogger, string path = null, Uri baseAddress = null)
         {
             _account = account;
             _containerName = containerName;
             _path = null;
-            _loggerFactory = loggerFactory;
+            _azureStorageLogger = azureStorageLogger;
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/AzureStorageFactory.cs]: just call this logger

File: src/NuGet.Services.Storage/FileStorageFactory.cs [hunk 4]
```diff
@@ -9,21 +9,21 @@ namespace NuGet.Services.Storage
     public class FileStorageFactory : StorageFactory
     {
         private readonly string _path;
-        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger<FileStorage> _fileStorageLogger;
 
-        public FileStorageFactory(Uri baseAddress, string path, ILoggerFactory loggerFactory)
+        public FileStorageFactory(Uri baseAddress, string path, ILogger<FileStorage> fileStorageLogger)
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/FileStorageFactory.cs]: just call this logger

**Pull request merged**